File: Main/static/JS/game/effects/stun.js
import { DurableEffect } from "./effect.js";
import { Modificator } from "../classes/modificator.js";

class Stun extends DurableEffect{
    constructor (hero, game_state, duration, caster){
        super('оглушение', 'stun.png', hero, game_state, duration, caster)
    }

    gen_description(){
        return `Оглушение[${this.duration}] (наложено ${this.caster.name})`
    }

    after_target_move(){
        this.decrease_duration()
    }

    gen_modificators(){
        return [new Modificator('stun', 1, true, 'after', 'set')]
    }
}

export {Stun}

File: Main/models/effects/stun.py
from .effect import SummaryDurableEffect
from django.db import models
from Main.models.modifier import Modifier


class Stun(SummaryDurableEffect):
    name = 'Stun'
    img_src = 'stun.png'

    def gen_description(self, link):
        return f"Оглушение ({self.duration} ходов) (наложено {link.caster.name})"

    def after_target_move(self, link):
        self.decrease_duration(link)

    def gen_modifiers(self, link):
        return [Modifier(hero=link.hero, param_name='stun', value=True, duration=1, is_set=True)]
